File: BreakOut3b/BOUser.m
//
//  BOUser.m
//  BreakOut3b
//
//  Created by Carl Tengwall on 2012-11-09.
//  Copyright (c) 2012 Carl Tengwall. All rights reserved.
//

#import "BOUser.h"

@implementation BOUser


@synthesize userId = _userId;
@synthesize difficulty = _difficulty;
@synthesize score = _score;
@synthesize backgroundColor = _backgroundColor;

// Constructor

- (id)initWithId:(NSString *) uid difficulty:(NSNumber *)diff score:(NSNumber *) aScore andBackground: (NSNumber *) bkgColor {
    self = [super init];
    
    if (self) {
        _userId = uid;
        _difficulty = diff;
        _score = aScore;
        _backgroundColor = bkgColor;
        //_ NSLog(@"bkgColor %@", bkgColor);
    }
    
    return self;
}

@end

File: BreakOut3b/BOUser.swift
import Foundation

@objc(BOUser)
final class BOUser: NSObject {
    @objc var userId: String
    @objc var difficulty: NSNumber
    @objc var score: NSNumber
    @objc var backgroundColor: NSNumber

    @objc(initWithId:difficulty:score:andBackground:)
    init(id: String, difficulty: NSNumber, score: NSNumber, backgroundColor: NSNumber) {
        self.userId = id
        self.difficulty = difficulty
        self.score = score
        self.backgroundColor = backgroundColor
        super.init()
    }
}
